validationparser/interceptor.go

```diff
@@ -0,0 +1,46 @@
+package validationparser
+
+import (
+	"context"
+	"reflect"
+	"strings"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type validator interface {
+	Validate() error
+}
+
+// UnaryServerInterceptor returns grpc.UnaryServerInterceptor
+// that should be used as a middleware that parser through the protoc-gen-validate middleware and returns a
+// user friendly error.
+func UnaryServerInterceptor() grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		if v, ok := req.(validator); ok {
+			if err := v.Validate(); err != nil {
```

Comment:
is this intended to replace the existing validation middleware we use? Otherwise, you're just calling validate again. A better approach would be some way of intercepting the response and checking for an error there

Reply:
that should just entail doing 
```
v, err := handler(ctx, req)
>> error checking/alteration stuff...
return v, err
```
instead of calling Validate manually and working with the error it returns.